sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1148,9 +1151,14 @@ def _retry_region_zones(self,
             CloudVmRayBackend().teardown_no_lock(handle,
                                                  terminate=need_terminate)
 
-        message = ('Failed to acquire resources in all regions/zones of '
-                   f'{to_provision.cloud}. '
-                   'Try changing resource requirements or use another cloud.')
+        if to_provision.zone is None:
+            message = ('Failed to acquire resources in all zones in '
+                       f'{to_provision.region}. Try changing resource '
+                       'requirements or use another region.')
```

Comment:
@Michaelvll Good and tough question. One possible exception is the local clusters. I didn't consider that.

Do you see any other possible exceptional case?

Reply:
I wondered whether the optimizer will always fill in the region for the `to_provision` for public cloud case. If it is true, I think it should be fine to keep the current way (I could not think of an exception).
nit: It would be nice if we can comment it in somewhere saying that `region` will always be filled by the optimizer for public cloud.

Reply:
I wondered whether the optimizer will always fill in the region for the `to_provision` for the public clouds. If it is true, I think it should be fine to keep the current way (I could not think of an exception).
nit: It would be nice if we could comment it somewhere saying that the optimizer will always fill `region` for the public clouds.

Reply:
Fixed. PTAL.